### PR TITLE
bounties: fee integration

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- Bounty Fee Integration
- Refund Fee Flow
- Metamask notification
- Ensure db isn't updated when user selects featured bounty and then rejects bounty

Check out [gist for info](https://gist.github.com/thelostone-mc/dd89b2824f1c5243fd256a902bb2b29b)

closes https://github.com/gitcoinco/web/issues/3891
closes https://github.com/gitcoinco/web/issues/3910
closes https://github.com/gitcoinco/web/issues/3909

Pending:  @frankchen07 

- email wiring (designs needed)
- enabling fee for increase bounty fee (design needed)

### CODE REVIEW

**Comment by frankchen07** [comment]
@danlipert @thelostone-mc 

First off, great work. Thanks for the amazing teamwork on this, and working through the decisions on contracts and stuff. +1 for Dan putting it up on staging 5 minutes after I asked for it too. Just finished looking at this on staging, and the functioning looks smooth. Some comments below from Aditya's asks.

1. We'll hold off on the email designs. Because this is a lightweight test, I would say we can remove the "request fee refund" option on the issue details page. If they decide to cancel, they can contact us via email / feedback for refunds and we can deal with it ad-hoc. Since we have the code for it already, we can always reactivate it when necessary. Our goal here is to gauge reactions and see what happens.

Screen Shot 2019-03-20 at 10 47 15

2. For the fee charge when increasing the bounty amount, I actually don't see an option to increase the bounty amount (example below). That being said, this is also an edge case that we don't need to worry about as of right now for the purposes of this revenue test. My hypothesis is that the majority of folks posting on Gitcoin aren't trying to game the system. 

Screen Shot 2019-03-20 at 11 07 53

3. I know we store featured bounty revenue in our DB, is the % fee also stored in our DB, and does it have a corresponding bounty_id that we can link to? This is mainly for data reporting purposes, as well as having the proper information to process refunds if necessary from our admin.

**Comment by thelostone-mc** [comment]
@frankchen07  

> actually don't see an option to increase the bounty amount (example below).

The increase bounty is the contribute button & I've seen it used by a few funders.
Click on contribute -> and then increase funding.
Every bounty has this option available to the bounty funder.
I've got the code nearly done but I can revert it if we still feel it's not needed 

>  is the % fee also stored in our DB, and does it have a corresponding bounty_id that we can link to?

we store the % fee directly in the bounty table

Screen Shot 2019-03-21 at 11 39 51 AM

File: app/assets/v2/css/submit_bounty.css [hunk 1]
```diff
@@ -146,6 +146,25 @@ textarea:read-only {
...
+  position: relative;
+  opacity: 0.12;
+  top: 2.2rem;
+}
+
+.fee-circle p {
+  color: #3E00FF;
+}
+
+#total-section {
```
Comment (octavioamu) [inline app/assets/v2/css/submit_bounty.css]: can we avoid id for styles? maybe `.border-top` can be useful here

File: app/dashboard/templates/bounty/process_refund_request.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+{% block breadcrumbs %}{% endblock %}
+
+{% block content %}
+<style>
+
+  body {
+    font-family: 'Muli';
+  }
+
+  #content {
```
Comment (octavioamu) [inline app/dashboard/templates/bounty/process_refund_request.html]: `.m-auto`

File: app/dashboard/templates/bounty/process_refund_request.html [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 0.95rem;
+    margin-top: 1rem;
+    margin-bottom: 1rem;
+  }
+
+  p {
+    font-weight: 600;
+  }
+
+  .section {
```
Comment (octavioamu) [inline app/dashboard/templates/bounty/process_refund_request.html]: `.mb-5`

File: app/dashboard/templates/bounty/process_refund_request.html [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    margin-bottom: 2rem;
+  }
+
+  textarea {
+    width: 100%;
+    height: 150px;
+    margin-bottom: 1rem;
+  }
+
+  .text-center {
```
Comment (octavioamu) [inline app/dashboard/templates/bounty/process_refund_request.html]: already exist in bootstrap

Comment (octavioamu) [reply app/dashboard/templates/bounty/process_refund_request.html]: wait is this page not using bootstrap?

File: app/assets/v2/css/submit_bounty.css (hunk 1 above)
Comment (thelostone-mc) [reply app/assets/v2/css/submit_bounty.css]: The styling is specific to that element. I could make it into a class but since this isn't used anywhere, would it give us any benefit? 🤔  
border-top doesn't give the color we need so I'd still have to add `border-color`.
Do we still wanna do that?

File: app/dashboard/templates/bounty/process_refund_request.html (hunk 2 above)
Comment (thelostone-mc) [reply app/dashboard/templates/bounty/process_refund_request.html]: This is on the admin view! 
Bootstrap styles aren't imported so `m-auto` would not work

File: app/dashboard/templates/bounty/process_refund_request.html (hunk 3 above)
Comment (thelostone-mc) [reply app/dashboard/templates/bounty/process_refund_request.html]: same as [here](https://github.com/gitcoinco/web/pull/3961/files#r269418938)

File: app/dashboard/templates/bounty/process_refund_request.html (hunk 4 above)
Comment (thelostone-mc) [reply app/dashboard/templates/bounty/process_refund_request.html]: same as [here](https://github.com/gitcoinco/web/pull/3961/files#r269418938)
Yup it is not using boostrap as this page is viewed on the django admin. Adding boostrap just for this would make it an overkill 😅
